### PR TITLE
Fix data pages embedding all image metadata

### PR DESCRIPTION
**Pull request by ikesau**
Fixes https://github.com/owid/owid-grapher/issues/3301

I remember taking a road trip that passed through a town with an odd bend in the highway as you exited. Apparently they'd started constructing the road from both ends, and only realized that they weren't aligned by the time they were in sight of each other.

image

This part of the datapage code feels a bit like that, except the two bits of road aren't even connected 😄 (All due respect to Daniel, who was under an enormous time crunch to get data pages implemented across the entire stack)

Changes:
- Removes the exhaustive `imageMetadataDicitionary` and other unused properties from FAQ documents that were being attached to the datapage's embedded JSON
- Attaches image metadata for the related research & writing's featured images (which are the only images we currently ned for datapages)
- renames `traverseEnrichedBlocks` → `traverseEnrichedBlock` (because it only takes a single block as an argument, even if that block contains multiple child blocks)

I decided not to refactor anything else to wait till https://github.com/owid/owid-grapher/issues/3426

### CODE REVIEW

**Comment by marcelgerber** [comment]
Nice storytelling in the PR description, would read again 😁

File: baker/GrapherBaker.tsx [hunk 1]
```diff
@@ -73,6 +71,15 @@ const renderDatapageIfApplicable = async (
...
     if (!variable) return undefined
 
+    // When baking from `bakeSingleGrapherChart`, we cache imageMetadata to avoid fetching every image for every chart
+    // But when rendering a datapage from the mockSiteRouter we want to be able to fetch imageMetadata on the fly
+    // And this function is the point in the two paths where it makes sense to do so
+    if (!imageMetadataDictionary) {
+        imageMetadataDictionary = await getAllImages(knex).then((images) =>
+            keyBy(images, "filename")
+        )
+    }
```
[the remark this answers is not in the document]
Comment (ikesau) [reply baker/GrapherBaker.tsx:81]: takes 0.1s 🙂

**Comment by ikesau** [comment]
@marcelgerber now's your time to read it again! 🙂 

We can go through it on a call together if you'd like

**Review by danyx23 — COMMENTED** [review]
Thanks a lot for cleaning this up after me! This looks good. I looked at http://localhost:3030/grapher/co-emissions-per-capita but the images don't show but this is probably because my local images aren't up to date - the paths seem right and the page size is smaller now.

I had a look at your staging server, at http://staging-site-datapage-image-attachments/grapher/co-emissions-per-capita , but that seems broken. Is that unrelated or is there an issue to look at still?

Otherwise it looks good to merge!

**Comment by danyx23** [comment]
Ah, I missed that this is already merged - thanks a lot :)

**Pull request merged**